src/main/java/com/launchdarkly/eventsource/EventSource.java

```diff
@@ -23,18 +20,18 @@
 
   private final URI uri;
   private final Headers headers;
-  private final ExecutorService executor;
+  private final ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
The `final` modifier prevents the executor from being `null`. The compiler will complain if there is a code path that does not initialize it.